app/controllers/runs_controller.rb

```diff
@@ -76,7 +76,7 @@ def download
 
     send_data(
       if params[:program] == @run.program.to_s
-        @run.file
+        @run.program.to_s == "llanfair" ? RunFile.pack_binary(@run.file) : @run.file
```

Comment:
I'd like to avoid adding filetype logic to the controller. Instead of deciding whether to pack here, it'd be nicer to put a method on `Run` like 

```ruby
def original_file
  program == :llanfair ? RunFile.pack_binary(file) : file
end
```

then just be able to call `@run.original_file` from here.